common/app/filters/RequestLoggingFilter.scala

```diff
@@ -1,59 +1,112 @@
 package filters
 
+import java.util.UUID
+
 import common.{ExecutionContexts, Logging, StopWatch}
-import play.api.mvc.{Result, RequestHeader, Filter}
+import play.api.mvc.{Filter, RequestHeader, Result}
 
 import scala.concurrent.Future
 import scala.util.{Failure, Random, Success}
 import conf.switches.Switches
+import net.logstash.logback.marker.LogstashMarker
+import net.logstash.logback.marker.Markers._
+import play.api.Logger
+
+import scala.collection.JavaConverters._
 
 object RequestLoggingFilter extends Filter with Logging with ExecutionContexts {
+
+  case class RequestLogger(rh: RequestHeader)(implicit internalLogger: Logger, stopWatch: StopWatch) {
+    private lazy val pseudoId = Random.nextInt(Integer.MAX_VALUE)
```

Comment:
`nextInt()` returns values between minInt and maxInt. `nextInt(n)` returns values between 0 and n
I wanted to avoid negative values here